httpbp/response_wrappers.go

```diff
@@ -0,0 +1,76 @@
+package httpbp
+
+import (
+	"net/http"
+)
+
+type optionalResponseWriter uint64
+
+const (
+	flusher optionalResponseWriter = 1 << iota
+	hijacker
+	pusher
+)
+
+func wrapResponseWriter(orig, wrapped http.ResponseWriter) http.ResponseWriter {
+	var f optionalResponseWriter
```

Comment:
nit: maybe consider using a different name for `f` so you don't have to make flusher one `fl` that's inconsistent with the others :)